### DISCUSSION TITLE
velero helm install fails

### DISCUSSION THREAD

**Discussion by thulasidassrinivasan** [vmware-tanzu/velero#3693] [Community support Q&A] [answered]
I am trying to install velero using helm in azure kubernetes service and i get below error message. 
Error: failed post-install: unable to build kubernetes object for deleting hook velero/templates/backupStorageLocation.yml: unable to recognize "": no matches for kind "BackupStorageLocation" in version "velero.io/v1"

command used to install 
helm install --generate-name --namespace velero -f charts/values-uat.yaml --set global.env=uat --set global.col=green charts/

Running velero backup-location get and velero snapshot-location get throws 
An error occurred: the server could not find the requested resource (get backupstoragelocations.velero.io)
An error occurred: the server could not find the requested resource (get volumesnapshotlocations.velero.io) 

velero version i am using is v1.3.2  and plugin version is velero/velero-plugin-for-microsoft-azure:v1.0.1. How to fix this ?

kubernetes version : 1.19.7

**Comment by zubron** [comment, marked as answer]
Hi @thulasidassrinivasan. This sounds like the CRDs for Velero have not been installed. Can you run `kubectl get crds` and verify that the output contains a number of CRDs with `velero.io` in the name?

I'm not an expert on Helm, but I wonder if the CRDs are not being installed due to the command you are using. Are there any other charts under your `charts/` directory? Please review our documentation on how to install the Helm chart as there are some differences that may impact how the resulting chart is installed: https://github.com/vmware-tanzu/helm-charts/tree/main/charts/velero#installing-the-velero-server

Thanks!

**Reply by thulasidassrinivasan** [reply]
@zubron - Thanks for replying. Below is the folder structure I have for helm deployment. 
![image](https://user-images.githubusercontent.com/55141977/115508621-d26e7c80-a27d-11eb-9464-5de3f8a22487.png)
Under crds/ I have below files
![image](https://user-images.githubusercontent.com/55141977/115508707-eade9700-a27d-11eb-9133-f303869252ba.png)
under templates/ I have below files. 
![image](https://user-images.githubusercontent.com/55141977/115508757-fb8f0d00-a27d-11eb-8dec-dbc23371b5b4.png)

contents of backupStorageLocation.yaml

`apiVersion: velero.io/v1
kind: BackupStorageLocation
metadata:
  name: '{{ .Values.configuration.backupStorageLocation.name }}'
  annotations:
    "helm.sh/hook": post-install,post-upgrade
    "helm.sh/hook-delete-policy": "before-hook-creation"
  labels:
    app.kubernetes.io/name: velero
    app.kubernetes.io/managed-by: Helm
    helm.sh/chart: velero-2.15.0
spec:
  provider: azure
  properties:
   objectStorage:
     bucket: '{{ .Values.global.col }}'
     prefix: '{{ .Values.global.col }}'
   config:
     resourceGroup: '{{ .Values.configuration.backupStorageLocation.config.resourceGroup }}'
     storageAccount: '{{ .Values.configuration.backupStorageLocation.config.storageAccount }}'
     subscriptionId: '{{ .Values.configuration.backupStorageLocation.config.subscriptionId }}`

**Reply by zubron** [reply]
@thulasidassrinivasan It looks like the version of the chart you are using is missing some of the CRDs necessary for it to function. Given that the CRDs aren't installed, I wonder if there was an error with your installation. If you can, I would suggest following the instructions for installing the chart here: https://github.com/vmware-tanzu/helm-charts/tree/main/charts/velero#installing-the-velero-server

**Reply by jenting** [reply]
@thulasidassrinivasan Which chart version you're using? and which helm version you're using?

**Reply by jenting** [reply]
Note that:
- currently Velero latest helm chart does not support helm v2 anymore.
- please run `helm repo update` to make sure that you're using the latest helm chart.

**Reply by thulasidassrinivasan** [reply]
Hi @jenting  I am using v3.5.3 of helm and I was trying to install velero v1.3.2. Because of tight deadlines I have to pause the usage of helm and move to velero cli. However I am not sure why I get this error message. As you can see I have all the files that is needed in crds and templates. Even though I have the of BackupStorageLocation as crds or as part of template I still get no matches for kind "BackupStorageLocation" in version "velero.io/v1".

**Reply by jenting** [reply]
- Which chart version you were using?
- Could you please try install CRDs by `velero install --crds-only --dry-run -o yaml | kubectl apply -f -` and then `helm install --skip-crds ...`?

**Reply by thulasidassrinivasan** [reply]
`apiVersion: v1
name: "velero"
description: "A helm chart"
version: 0.1.0`

ok i will give crds installation alone and let you know. Thanks for replying

**Reply by jenting** [reply]
- Why the version is so old `0.1.0`?
- the `apiVersion: v1` means it's compatible with helm v2.

Are you using the chart from https://github.com/vmware-tanzu/helm-charts?

**Reply by thulasidassrinivasan** [reply]
Ah that might be the problem.  I think I have to use below charts and bump the velero version also to 1.6.0. I will give this a try today. 

`apiVersion: v2
appVersion: 1.6.0
description: A Helm chart for velero
name: velero
version: 2.19.2`

**Reply by jenting** [reply]
You could use the chart and pass `--set image.tag=v1.3.2` to overwrite the default container image `v1.6.0`.
The upgrade CRDs job would help you install the Velero v1.3.2 compatible CRDs.

**Reply by thulasidassrinivasan** [reply]
I get the warning. Anything I have to worry about ?

`W0429 09:16:57.841244   24577 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition`

**Reply by jenting** [reply]
No worries, this is because the API version `apiextensions.k8s.io/v1beta1` is deprecated and will be unavailable in v1.22+.

**Reply by thulasidassrinivasan** [reply]
Thanks but I could see all crd's are still using `apiextensions.k8s.io/v1beta1` even if it install it via velero cli.  When it is planned to be changed ?

**Reply by jenting** [reply]
Velero has to cover multiple K8s version >= 1.12, this means that Velero needs to support both `apiextensions.k8s.io/v1beta1` and `apiextensions.k8s.io/v1` simultaneously. Furthermore, since Velero supports displaying CRDs only (`velero install --crds-only --dry-run -o yaml`), we don't know which K8s version the user is using now.
It's not an intuition just bump the version from `v1beta` to `v1`.

https://github.com/vmware-tanzu/velero/issues/2505#issuecomment-809652562

**Reply by jenting** [reply]
@thulasidassrinivasan Is the issue be solved?

**Reply by thulasidassrinivasan** [reply]
@jenting - Yeah. Thanks for the answers.
